From 3b02ca3218cf878e206545041aa811073e4f6c79 Mon Sep 17 00:00:00 2001
From: Alan Stern <stern@rowland.harvard.edu>
Date: Fri, 30 Apr 2010 12:42:23 -0400
Subject: USB: export the new ch11.h file to userspce
Patch-mainline: v2.6.35
Git-commit: 3b02ca3218cf878e206545041aa811073e4f6c79

This patch (as1374) cleans up a few loose ends in the
include/linux/usb/ch11.h header file and exports it to userspace.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Cc: Eric Lescouet <Eric.Lescouet@virtuallogix.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/usb/Kbuild |    1 +
 include/linux/usb/ch11.h |   10 ++++------
 2 files changed, 5 insertions(+), 6 deletions(-)

--- a/include/linux/usb/Kbuild
+++ b/include/linux/usb/Kbuild
@@ -1,6 +1,7 @@
 header-y += audio.h
 header-y += cdc.h
 header-y += ch9.h
+header-y += ch11.h
 header-y += gadgetfs.h
 header-y += midi.h
 header-y += g_printer.h
--- a/include/linux/usb/ch11.h
+++ b/include/linux/usb/ch11.h
@@ -6,12 +6,10 @@
  * drivers.
  */
 
-#ifndef __LINUX_HUB_H
-#define __LINUX_HUB_H
+#ifndef __LINUX_CH11_H
+#define __LINUX_CH11_H
 
-#include <linux/list.h>
-#include <linux/workqueue.h>
-#include <linux/compiler.h>	/* likely()/unlikely() */
+#include <linux/types.h>	/* __u8 etc */
 
 /*
  * Hub request types
@@ -166,4 +164,4 @@ enum hub_led_mode {
 #define HUB_TTTT_24_BITS	0x40
 #define HUB_TTTT_32_BITS	0x60
 
-#endif /* __LINUX_HUB_H */
+#endif /* __LINUX_CH11_H */
